### PR TITLE
Add new version OpenSSL 1.1.1 to Travis CI.

### PR DESCRIPTION
**Pull request by junaruga**
This PR is to enable the OpenSSL latest stable version 1.1.1 on Travis CI GCC latest version gcc-8 test case. Current used OpenSSL version is 1.0.1f.
I am going to open the ticket on Ruby bug tracking system soon.

The discussion can be in https://bugs.ruby-lang.org/issues/15220 .

### CODE REVIEW

File: tool/install_openssl.sh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+pushd "${TMP_DIR}/openssl-${OPENSSL_VERSION}"
+if [ -d "${OPENSSL_DIR}" ]; then
+  rm -rf "${OPENSSL_DIR}"
+fi
+./Configure \
+  --prefix=${OPENSSL_DIR} \
+  enable-crypto-mdebug enable-crypto-mdebug-backtrace \
+  linux-x86_64
+make -s $JOBS
+make -s install_sw
```
Comment (junaruga) [inline tool/install_openssl.sh]: This build logic is inspired from `ruby-openssl-docker` image's `Dockerfile`.
https://github.com/ruby/openssl/blob/master/tool/ruby-openssl-docker/Dockerfile

File: .travis.yml [hunk 2]
```diff
@@ -21,6 +21,10 @@ dist: trusty
 
 sudo: false
 
+cache:
+  - directories:
+    - $HOME/openssl
+
```
Comment (junaruga) [inline .travis.yml]: Sorry, those lines might be wrong. We might not have to add those lines.
I referred python's case [1]. But reading the cache document [2], it is still not clear for me.

* [1] https://github.com/python/cpython/blob/master/.travis.yml#L11
* [2] https://docs.travis-ci.com/user/caching

**Comment by junaruga** [comment]
I added new commit to enable to install OpenSSL 1.1.1 for only Travis cron mode that is currently maybe run daily. That's to save the total running time.
I think that it's still okay to detect an issue related to new OpenSSL as early as possible.

**Comment by shyouhei** [comment]
Closing. https://bugs.ruby-lang.org/issues/15220 is also closed.

**Pull request closed without merging**